### PR TITLE
Document colcon_defaults.yaml feature

### PR DESCRIPTION
**Pull request by jdlangs**
Documentation for colcon/colcon-defaults#29

### CODE REVIEW

**Review by cottsay — APPROVED** [review]

**Comment by giuliano-97** [comment]
hi, I spent quite some time implementing good workarounds to achieve what this feature does just to find out after adventuring in the rabbit hole of github issues that this feature already exists - how about merging this so that the documentation is up to date and everyone can benefit from this? 😄

File: user/configuration.rst [hunk 1]
```diff
@@ -94,9 +94,16 @@ The following are a few examples (see e.g. ``colcon build --help``):
...
+Default files are used to configure ``colcon`` behavior for all packages being operated on.
+Multiple configuration files can be used to modify verb behavior without using the CLI:
 
+* If the configuration file ``$COLCON_HOME/defaults.yaml`` exists it is used to customize the default behavior of the CLI.
+  The location can also be modified using the environment variable ``COLCON_DEFAULTS_FILE`` (see ``colcon --help``).
+
+* If the configuration file ``colcon_defaults.yaml`` exists in a workspace root, it will be used to change the behavior of any invocation of ``colcon`` in that workspace.
+  The option values specified in the workspace file will always override the global defaults.
+
+Configuration files use standard YAML syntax.
```
[the remark this answers is not in the document]
Comment (jdlangs) [reply user/configuration.rst]: Sounds good, committed the suggested change.

**Comment by jdlangs** [comment]
@giuliano-97 that's my bad for forgetting about this PR. It should be good to go. The failing readthedocs check looks like something new that probably should be resolved in another PR.

**Comment by cottsay** [comment]
I'm working on the failing deployment.

**Comment by sdmcgrath** [comment]
@jdlangs I think this should pass if you update past https://github.com/colcon/colcon.readthedocs.org/pull/99

**Comment by jdlangs** [comment]
> @jdlangs I think this should pass if you update past #99

Thanks! Looking good now.

**Comment by roncapat** [comment]
Would it be possible to merge? I was searching for this piece of info online and it would have been very useful to find it in the official docs.

**Comment by giuliano-97** [comment]
@jdlangs @cottsay

**Pull request merged**